src/integrationTest/java/org/opensearch/security/privileges/PrivilegesEvaluatorTest.java

```diff
@@ -68,4 +72,43 @@ public void testRegexPattern() throws Exception {
         }
 
     }
+
+    @Test
+    public void testSearchTemplateRequestSuccess() {
+        try (TestRestClient client = cluster.getRestClient(SEARCH_TEMPLATE)) {
+            assertThat(
+                client.getWithJsonBody(
+                    "services/_search/template",
+                    "{\"source\":{\"query\":{\"match\":{\"service\":\"{{service_name}}\"}}},\"params\":{\"service_name\":\"Oracle\"}}"
+                ).getStatusCode(),
+                equalTo(HttpStatus.SC_OK)
+            );
+        }
+    }
+
+    @Test
+    public void testSearchTemplateRequestUnauthorizedIndex() {
+        try (TestRestClient client = cluster.getRestClient(SEARCH_TEMPLATE)) {
+            assertThat(
+                client.getWithJsonBody(
+                    "movies/_search/template",
+                    "{\"source\":{\"query\":{\"match\":{\"service\":\"{{service_name}}\"}}},\"params\":{\"service_name\":\"Oracle\"}}"
```

Comment:
Nit; you can refactor out the request body since they are the same, should clean up the code

Reply:
Addressed here: [c4a3d5d](https://github.com/opensearch-project/security/pull/2921/commits/c4a3d5ddacc84a678e762d2402339464aa9348e8), that is better, thanks!